www/perl-lib/SMT/Utils.pm

```diff
@@ -127,6 +129,46 @@ sub db_connect
     return $dbh;
 }
 
+=item hasRegSharing()
+
+Check whether registration sharing is enabled or not.
+
+=cut
+
+sub hasRegSharing
+{
+    my $r = shift;
+
+    if (! defined $REGSHARING) {
+        $REGSHARING = 0;
+        my $cfg;
+        eval
+        {
+            $cfg = SMT::Utils::getSMTConfig();
+        };
+        if($@ || !defined $cfg)
+        {
+            if (! $r)
+            {
+                return;
+            }
+            $r->log_error("Cannot read the SMT configuration file: ".$@);
+            my $msg = 'SMT server is missconfigured. Please contact your '
+                . 'administrator.';
+            return http_fail($r, 500, $msg);
```

Comment:
the return value of http_fail is an Apache2::Const::OK; which evaluate to "true". This would
indicate, that the loading of the module was successfull.
I would suggest to re-write this to:

```
  if ($r)
  {
      $r->log_error("Cannot read the SMT configuration file: ".$@);
  }
  return;
```